config/pmix.m4

```diff
@@ -683,6 +683,29 @@ AC_DEFUN([PMIX_SETUP_CORE],[
           [AC_DEFINE_UNQUOTED([HAVE_UNIX_BYTESWAP], [1],
                               [whether unix byteswap routines -- htonl, htons, nothl, ntohs -- are available])])
 
+    #
```

Comment:
From my local testing, the `AC_CHECK_FUNCS([pthread_setaffinity_np])` test is sufficient, with one minor change.  You need to move the check until after the call to `PMIX_CONFIG_THREADS`.  If there's a platform where `pthread_setaffinity_np()` is a compiler builtin instead of a library function (and I doubt such a place exists), then `AC_CHECK_FUNCS` would fail.  But the problem here was just that the lack of the `-pthread` or `-lpthread` (depending on platform) meant that there was no library providing `pthread_setaffinity_np` in the link command.

Moving `AC_CHECK_FUNCS([pthread_setaffinity_np])` to below the call to `PMIX_CONFIG_THREADS` resulted in a "yes" on modern Linux and a "no" on MacOS.

Reply:
Verified it works - thx!